DigitalLearningSolutions.Web/Controllers/TrackingSystem/Centre/Administrator/AdministratorController.cs

```diff
@@ -131,6 +131,47 @@ public IActionResult UnlockAccount(int adminId)
             return RedirectToAction("Index");
         }
 
+        [Route("{adminId:int}/DeactivateAdmin")]
+        [HttpGet]
+        public IActionResult DeactivateAdmin(int adminId)
+        {
+            var centreId = User.GetCentreId();
+            var adminUser = userDataService.GetAdminUserById(adminId);
+            if (adminUser == null || adminUser.CentreId != centreId)
+            {
+                return NotFound();
+            }
+
+            var model = new DeactivateAdminViewModel(adminUser);
+            return View(model);
+        }
+
+        [Route("{adminId:int}/DeactivateAdmin")]
+        [HttpPost]
+        public IActionResult DeactivateAdmin(int adminId, DeactivateAdminViewModel model)
+        {
+            var centreId = User.GetCentreId();
+            var adminUser = userDataService.GetAdminUserById(adminId);
+
+            if (!model.Confirm)
+            {
+                ModelState.AddModelError(
+                    nameof(DeactivateAdminViewModel.Confirm),
+                    "You must confirm before deactivating this account."
+                );
+
+                return View(model);
+            }
+
+            if (adminUser == null || adminUser.CentreId != centreId)
+            {
+                return NotFound();
+            }
+
+            userDataService.DeactivateAdmin(adminId);
+            return View("DeactivateAdminConfirmation", adminUser);

```

Comment:
The adminUser here is not actually being used for anything as the View does not expect a model to be passed in, so it can be safely removed from this call.

Reply:
This is used in test method to check the status when it's changed. Please see- DeactivateAdminUser_deactivate_admin_user_with_confirmation

Reply:
To clarify, the adminUser returned here is the same one that is retrieved from the call to `userDataService.GetAdminUserById(adminId)` on line 154. That adminUser will have Active as true, and that variable won't be changed as it is just a copy of whatever the data in the DB was when it was created.  As such, this returned adminUser is not the updated adminUser, so this isn't actually testing if it has been deactivated.

In fact, that is all irrelevant as we don't _need_ to test that this method actually deactivates the admin. The test to see if an admin is deactivated is already done in the data service tests. We actually can't test that the controller deactivates the admin, as the DataService used in the AdminControllerTests is a mock object, and so it won't actually deactivate the admin. Simply testing that a call is made to the DataService method is adequate.